src/spikeinterface/preprocessing/whiten.py

```diff
@@ -162,7 +187,7 @@ def compute_whitening_matrix(recording, mode, random_chunk_kwargs, apply_mean, r
 
     """
     random_data = get_random_data_chunks(recording, concatenated=True, return_scaled=False, **random_chunk_kwargs)
-    random_data = random_data.astype("float32")
+    random_data = random_data
```

Comment:
Why did you removed this ? When data is integer we we float no ?

Reply:
I think it's ok, but we can then remove the line:
```suggestion
```